datafusion/functions/src/datetime/to_timestamp.rs

```diff
@@ -240,8 +254,13 @@ impl ScalarUDFImpl for ToTimestampMillisFunc {
         &self.signature
     }
 
-    fn return_type(&self, _arg_types: &[DataType]) -> Result<DataType> {
-        Ok(Timestamp(Millisecond, None))
+    fn return_type(&self, arg_types: &[DataType]) -> Result<DataType> {
```

Comment:
may be we can factor this out? the same block is too repetitive

Reply:
I agree -- I made a follow on PR here: https://github.com/apache/datafusion/pull/11116